src/rpc/util.h

```diff
@@ -414,9 +414,7 @@ class RPCHelpMan
      * argument isNull() and parses (from JSON) and returns the user-passed argument,
      * or the default value derived from the RPCArg documentation.
      *
-     * There are two overloads of this function:
-     * - Use Arg<Type>(size_t i) to get the argument (or the default value) by index.
-     * - Use Arg<Type>(const std::string& key) to get the argument (or the default value) by key.
+     * Use Arg<Type>(const std::string& key) to get the argument (or the default value) by key.
```

Comment:
nit: this is just the brief, so should be removed/merged imo (+ for MaybeArg)

Reply:
Ah good point, I hadn't considered that the `Type` alias is introduced here. I think a preferable approach to me is to use the actual `typename` instead of aliasing it, such as in this diff. Optionally, I'd also be supportive of renaming `R` to something more descriptive such as `ReturnType`, but even with just `R` I think this is more clear:

<details>
<summary>git diff on fadb3eb57b</summary>

```diff
diff --git a/src/rpc/util.h b/src/rpc/util.h
index 9ba7fcf5e6..8fc598b873 100644
--- a/src/rpc/util.h
+++ b/src/rpc/util.h
@@ -414,11 +414,9 @@ public:
      * argument isNull() and parses (from JSON) and returns the user-passed argument,
      * or the default value derived from the RPCArg documentation.
      *
-     * Use Arg<Type>(const std::string& key) to get the argument (or the default value) by key.
+     * The instantiation of this helper for type R must match the corresponding RPCArg::Type.
      *
-     * The Type passed to this helper must match the corresponding RPCArg::Type.
-     *
-     * @return The value of the RPC argument (or the default value) cast to type Type.
+     * @return The value of the RPC argument (or the default value) cast to type R.
      *
      * @see MaybeArg for handling optional arguments without default values.
      */
@@ -446,12 +444,10 @@ public:
      * argument isNull() and parses (from JSON) and returns the user-passed argument,
      * or a falsy value if no argument was passed.
      *
-     * Use MaybeArg<Type>(const std::string& key) to get the optional argument by key.
-     *
-     * The Type passed to this helper must match the corresponding RPCArg::Type.
+     * The instantiation of this helper for type R must match the corresponding RPCArg::Type.
      *
-     * @return For integral and floating-point types, a std::optional<Type> is returned.
-     *         For other types, a Type* pointer to the argument is returned. If the
+     * @return For integral and floating-point types, a std::optional<R> is returned.
+     *         For other types, a R* pointer to the argument is returned. If the
      *         argument is not provided, std::nullopt or a null pointer is returned.
      *
      * @see Arg for handling arguments that are required or have a default value.

```
</details>